_test/test_sqw_class/test_sqw_copy.m

```diff
@@ -28,19 +28,36 @@ function test_copy_returns_distinct_object(obj)
             sqw_copy = copy(sqw_obj);
             
             sqw_copy.main_header.title = 'test_copy';
-            sqw_copy = sqw_copy.change_header(struct([]));
+            assertFalse(equal_to_tol(sqw_copy.main_header, sqw_obj.main_header));
+            
+            % want to test expinfo/instruments and expinfo/detpar
+            % separately, so make 2 copies of expinfo
+            expinf1 = sqw_copy.experiment_info;
+            expinf2 = sqw_copy.experiment_info;
+
+            % all instruments changed with new name to make expinf different
+            for i=1:numel(expinf1.instruments), expinf1.instruments{i}.name = 'copy'; end
```

Comment:
Personally, I really dislike inline `do`s/`if`s like this. Definitely a personal matter.

Reply:
```matlab
expinf1.instruments = cellfun(@(x) setfield(x, 'name', 'copy'), expinf1.instruments, 'UniformOutput', false)
``` 
might be cleaner? Though admittedly MATLAB's verbosity in making the output a cell-array this sort of defeats that.

Reply:
Happy to change the formatting. Arguable if the cellfun version is cleaner - it is very opaque and in the interests of making clear what a test does if there is the problem here I would retain the for loop.